griptape/structures/structure.py

```diff
@@ -133,8 +134,8 @@ def resolve_relationships(self) -> None:
                 if task.id not in child.parent_ids:
                     child.parent_ids.append(task.id)
 
-    @observable
     def before_run(self, args: Any) -> None:
+        super().before_run(args)
```

Comment:
Want to make sure I understand (1) the implications of moving `@observable` from here to the parent (like will this still instrument the child class like before?) and (2) why did you get the `TypeError` originally which prompted the change?

Reply:
For (1), I think the unit tests have me satisfied half satisfied. I'd like one empirical test to ensure that the child behavior is fully wrapped (do exceptions propogate? is timing accurate?).

For (2), I have a hunch its because of the use of generics, we have `self: T` in the parent class. When you call `super().before_run(args)`, is the type of `self`, really `T`? I wouldn't expect so. I have a funny feeling about `self: T`

Reply:
Woohoo we solved this offline by being explicit about which parent we're calling